lib/pure/rationals.nim

```diff
@@ -308,6 +329,7 @@ func floorMod*[T: SomeInteger](x, y: Rational[T]): Rational[T] =
 
 func hash*[T](x: Rational[T]): Hash =
   ## Computes the hash for the rational `x`.
+  if x.num == 0: return hash toRational 0
```

Comment:
```suggestion
  if x.num == 0: return hash(toRational(0))
```
Especially for nested function calls, I find command syntax very unreadable.

Reply:
Especially for nested function calls, I find using parenthesis quite messy and less readable.
Compare
```nim
sum list.mapIt it.len.f
```
to

```nim
sum(mapIt(list, f(len(it))))
```